src/Control/Foldl.hs

```diff
@@ -1117,20 +1119,20 @@ premap f (Fold step begin done) = Fold step' begin done
 {-| @(premapM f folder)@ returns a new 'FoldM' where f is applied to each input
     element
 
-> foldM (premapM f folder) list = foldM folder (map f list)
+> foldM (premapM f folder) list = mapM f list >>= foldM folder
```

Comment:
Technically this is not correct.  I would just remove this example since there is no good equivalent translation in terms of another utility

Reply:
Whoops, yeah. The monadic effects are performed in different orders.